safety/scan/command.py

```diff
@@ -175,6 +186,101 @@ def process_report(
     return report_url
 
 
+def filter_json_keys(json_string: str, keys: List[str]) -> str:
+    """
+    Filters the given JSON string by the specified top-level keys.
+
+    Args:
+        json_string (str): The JSON string to filter.
+        keys (List[str]): List of top-level keys to include in the output.
+
+    Returns:
+        str: A JSON string containing only the specified keys.
+    """
+    report_dict = json.loads(json_string)
+    filtered_data = {key: report_dict[key] for key in keys if key in report_dict}
+    return json.dumps(filtered_data, indent=4)
+
+
+def filter_valid_cves(vulnerabilities: List[Any]) -> List[Dict[str, Any]]:
+    """
+    Filters and returns valid CVE details from a list of vulnerabilities.
+
+    Args:
+        vulnerabilities (List[Any]): A list of vulnerabilities, which may include invalid data types.
+
+    Returns:
+        List[Dict[str, Any]]: A list of filtered CVE details that are either strings or dictionaries.
+    """
+    return [
+        cve for cve in vulnerabilities if isinstance(cve, str) or isinstance(cve, dict)
+    ]  #type:ignore
+
+
+def sort_cve_data(cve_data: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """
+    Sorts CVE details by severity in descending order.
+
+    Args:
+        cve_data (List[Dict[str, Any]]): A list of CVE details dictionaries, each containing a 'severity' key.
+
+    Returns:
+        List[Dict[str, Any]]: The sorted list of CVE details, prioritized by severity (e.g., CRITICAL > HIGH > MEDIUM).
+    """
+    severity_order = {
```

Comment:
nit: We can probably relay on VulnerabilitySeverityLabels and do something like:

severity_order = {key.name: id for (id, key) in enumerate(VulnerabilitySeverityLabels)}
sorted(cve_data, key=lambda x: severity_order.get(x["severity"].upper(), 0), reverse=True)

Reply:
Ah switched now